keras/backend/theano_backend.py

```diff
@@ -1144,7 +1144,18 @@ def spatial_2d_padding(x, padding=((1, 1), (1, 1)), data_format=None):
                    py_slice(left_pad, input_shape[2] + left_pad),
                    py_slice(None))
     y = T.set_subtensor(output[indices], x)
-    y._keras_shape = output_shape
```

Comment:
The keras shape is not correct, actually the test(```backend_test.py/test_spatial_2d_padding```) already cover this case, but due to the bug mentioned/fixed in this PR, it doesn't complain.